src/Microsoft.AspNet.JsonPatch/Adapters/IObjectAdapter.cs

```diff
@@ -7,14 +7,13 @@ namespace Microsoft.AspNet.JsonPatch.Adapters
 {
     /// <summary>
     /// Defines the operations that can be performed on a JSON patch document.
-    /// </summary>
-    /// <typeparam name="TModel">The type of the model.</typeparam>
-    public interface IObjectAdapter<TModel> where TModel : class
+    /// </summary>  
+    public interface IObjectAdapter    
```

Comment:
Why dynamic? Did you compare the before and after performance impact of this change?